dev/com.ibm.ws.jbatch.security/src/com/ibm/ws/jbatch/security/impl/WSBatchAuthServiceImpl.java

```diff
@@ -336,30 +433,42 @@ public boolean isMonitor() {
         return this.isMonitor(runAsSubject());
     }
 
+    @Override
+	public boolean isGroupMonitor() {
+    	return this.isGroupMonitor(runAsSubject());
+	}
+    
     @Override
     public boolean isInAnyBatchRole() {
         return this.isInAnyBatchRole(runAsSubject());
     }
 
     @Override
+    @FFDCIgnore(JobSecurityException.class)
     public boolean isAuthorizedInstanceRead(long instanceId)
         throws NoSuchJobInstanceException {
-        String submitter = persistenceManagerService.getJobInstanceSubmitter(instanceId);
-
-        if (this.isAdmin(runAsSubject())) {
-        } else if (this.isMonitor(runAsSubject())) {
-        } else if (batchSecurityHelper.getRunAsUser().equals(submitter)) { 
-            if (!this.isInAnyBatchRole(runAsSubject())) {
-                //We could just check for && isSubmittter, because we already checked for admin or monitor,
-                //but logically as long as you are in any batch role you can view your own jobs
-                return false;
-            }
-        } else {
-            logger.finer("Current user " + this.getRunAsUser()+ " does not match the tag of record");
-            return false;
-        }
+    	    	
+    	try {
+    		authorizedInstanceRead(instanceId);
+    	}
+    	catch (JobSecurityException jse){
+    		return false;
+    	}
+    	
+    	return true;
+    }
 
-        return true;
+	private boolean subjectInGroups(Set<String> listOfGroupsForSubject, Set<String> listOfGroupsForJobID) {
+		
+		Iterator it = listOfGroupsForSubject.iterator();
+		while (it.hasNext()){
+			if (listOfGroupsForJobID.contains(it.next())){
+				// found a group match, good to go
+				return true;
```

Comment:
We're not even logging which group matched?  Probably would typically be obvoius... but wouldn't hurt to log each iteration through with the name and fail-or-pass